library/ssl_client.c

```diff
@@ -790,9 +792,11 @@ static int ssl_prepare_client_hello( mbedtls_ssl_context *ssl )
      * RFC 5077 section 3.4: "When presenting a ticket, the client MAY
      * generate and include a Session ID in the TLS ClientHello."
      */
+        int renegotiating = 0;
 #if defined(MBEDTLS_SSL_RENEGOTIATION)
-        if( ssl->renego_status == MBEDTLS_SSL_INITIAL_HANDSHAKE )
+        renegotiating = ( ssl->renego_status != MBEDTLS_SSL_INITIAL_HANDSHAKE );
```

Comment:
Minor: here and elsewhere, I think keeping the `if` (as e.g. in `mbedtls_ssl_read_record`) would make the code easier to read.
```
int foo = 0;
#if FEATURE
if( preexisting condition )
    foo = 1;
#endif
if( foo )
```